fuse_optimizers/test/launch_tests/test_fixed_lag_ignition.cpp

```diff
@@ -31,58 +31,95 @@
  *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  *  POSSIBILITY OF SUCH DAMAGE.
  */
-#include <fuse_models/SetPose.h>
-#include <geometry_msgs/PoseWithCovarianceStamped.h>
-#include <nav_msgs/Odometry.h>
-#include <ros/ros.h>
+#include <fuse_core/time.hpp>
+#include <fuse_msgs/srv/set_pose.hpp>
+#include <geometry_msgs/msg/pose_with_covariance_stamped.hpp>
+#include <nav_msgs/msg/odometry.hpp>
+#include <rclcpp/rclcpp.hpp>
+#include <rclcpp/wait_for_message.hpp>
+#include <std_srvs/srv/empty.hpp>
 
 #include <gtest/gtest.h>
 
 
-TEST(FixedLagIgnition, SetInitialState)
+class FixedLagIgnitionFixture : public ::testing::Test
 {
-  // TODO(CH3): Make this an rclcpp node
-  auto node_handle = ros::NodeHandle();
-  auto relative_pose_publisher = node_handle.advertise<geometry_msgs::PoseWithCovarianceStamped>("/relative_pose", 1);
+public:
+  FixedLagIgnitionFixture()
+  {
+  }
+
+  void SetUp() override
+  {
+    executor_ = std::make_shared<rclcpp::executors::SingleThreadedExecutor>();
+    spinner_ = std::thread(
+      [&]() {
+        executor_->spin();
+      });
+  }
+
+  void TearDown() override
+  {
+    executor_->cancel();
+    if (spinner_.joinable()) {
+     spinner_.join();
+    }
+    executor_.reset();
+  }
+
+   std::thread spinner_;  //!< Internal thread for spinning the executor
+   rclcpp::executors::SingleThreadedExecutor::SharedPtr executor_;
+};
+
+TEST_F(FixedLagIgnitionFixture, SetInitialState)
+{
+  auto node = rclcpp::Node::make_shared("fixed_lag_ignition_test");
+  executor_->add_node(node);
+
+  auto relative_pose_publisher =
+    node->create_publisher<geometry_msgs::msg::PoseWithCovarianceStamped>(
+      "pose_sensor/relative_pose", 1);
```

Comment:
In the ROS 1 code, the pose topic for the pose publisher uses a [private_node_handle](https://github.com/locusrobotics/fuse/blob/devel/fuse_publishers/src/pose_2d_publisher.cpp#L173), so it should be prepended with the namespace of the node.

But in the port, we're using the node of the optimizer (and we're also not using subnodes), so this prepending functionality has been replaced with [joinTopicName](https://github.com/locusrobotics/fuse/blob/rolling/fuse_publishers/src/pose_2d_publisher.cpp#L198))

I just mimicked the same sort of behavior with the test. (In this case the test is just bringing up a bare publisher instead of a fuse publisher.)